apps/client/src/routes/routes.test.tsx

```diff
@@ -0,0 +1,80 @@
+import { describe, it, expect } from 'vitest';
+import { fireEvent, render, screen } from '@testing-library/react';
+import { createMemoryRouter, RouterProvider } from 'react-router-dom';
+import routes  from './routes';
+
+const renderRouter = (route: string) => {
+  const router = createMemoryRouter(routes, {
+    initialEntries: [route],
+  });
+
+  render(<RouterProvider router={router} />);
+}
+
+describe('dashboard', () => {
+  describe('list', () => {
+    it('renders', async () => {
+      renderRouter('/dashboards');
+
+      expect(screen.getByRole('heading')).toHaveTextContent('Dashboards')
+
+      expect(screen.getAllByRole('link')[0]).toHaveTextContent('Dashboard 1');
```

Comment:
I don't think this is the right place to make assertions about the dashboard. It seems like the correct assertion would be to assert on the URL after requesting route change.

Reply:
Thanks for bringing this up! I think I disagree on this one. What's more important IMO is that the page renders the things we want on the page. We already know what route we are on in line 17.

Reply:
This is also the recommended way of testing routes https://testing-library.com/docs/example-react-router/

Reply:
Thank you for sharing the resource! We should store this link in the code.

In their recommendation, that's an `app.test.ts` file, which may be the appropriate place to test these things in a small app.

IMO we should keep things that change at the same time in the same place. Having a `route.test.ts` file making assertions on page implementation details is going to leave future engineers frustrated. If you are modifying a page and not adding a route, you will not expect to change `routes.test.ts` until you see a failing test. Worst case scenario is this file gets neglected as assertions slip by.

I recommend refactoring the tests by putting the assertions about the pages within the individual page test suites.

Reply:
I like that idea. Let me refactor this accordingly.

Reply:
Thank you @NorbertNader!